.github/workflows/validate-csv.yml

```diff
@@ -0,0 +1,30 @@
+name: validate-csv
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+
+
+jobs:
+  validate-csv:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v3
+
+      - name: Install Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install requirements
+        run: pip install requests pandas numpy
+
+      - name: Set up output matcher for PR annotations
+        run: echo '::add-matcher::.github/workflows/validate-csv-matcher.json'
```

Comment:
Can you explain a bit about what happens here? This doesn't seem like Python 😆

Reply:
This is using a feature of GitHub Actions to enable automatic annotations/comments in the pull requests's file view, like this:

![image](https://github.com/AssessingSolar/solarstations/assets/57452607/8d79ac57-c79a-410b-9997-0091f28e64fb)

Basically:
- Some part of the CI job emits logging output containing information to be used for annotations: https://github.com/AssessingSolar/solarstations/pull/143/files#diff-0dcf874f7ba8617229c0b2282660a0d73587408a21c4ccefac267ea348cc385eR75-R76
- The matcher.json file defines a regex that identifies those special log output lines and extracts the necessary annotation information (which file, which line, what message to display in the annotation): https://github.com/AssessingSolar/solarstations/pull/143/files#diff-7cd4b2d9ed8f98f39437ec6dbc12a17d4fd790a14daf6c6c546f4aca810941beR8-R12
- This `echo '::add-matcher::.github/workflows/validate-csv-matcher.json'` line enables the specified matching+annotations for this CI job